### PR TITLE
Bilateral pinching with PinchDial

### PR DESCRIPTION
**Pull request by Brendonovich**
Title says it all

### CODE REVIEW

File: Apollo/Components/Dial.xaml [hunk 1]
```diff
@@ -6,7 +6,8 @@
   <Grid RowDefinitions="Auto,Auto,Auto" VerticalAlignment="Center" LayoutUpdated="LayoutChanged">
     <TextBlock Grid.Row="0" HorizontalAlignment="Center" x:Name="Title" />
 
-    <Canvas Grid.Row="1" Margin="0 5" HorizontalAlignment="Center" VerticalAlignment="Center" Width="43" Height="39" Background="Transparent" Cursor="Hand" x:Name="ArcCanvas"
+    <Canvas Grid.Row="1" Margin="5 0" HorizontalAlignment="Center" VerticalAlignment="Center" Width="43" Height="39"
```
Comment (mat1jaczyyy) [inline Apollo/Components/Dial.xaml]: why was margin changed here?

File: Apollo/Components/GraphDial.cs [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+using Apollo.Structures;
+using Avalonia.Media.Imaging;
+using Avalonia.Platform;
+
+namespace Apollo.Components {
+    public class GraphDial: Dial {
+        
+        public delegate void DialBilateralChangedEventHandler(GraphDial sender, bool NewValue, bool? OldValue);
+        public event DialBilateralChangedEventHandler BilateralChanged;
+
```
Comment (mat1jaczyyy) [inline Apollo/Components/GraphDial.cs]: why do you need this event? can't you reuse ModeChanged with the invoke func?

File: Apollo/Components/GraphDial.cs [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+            }
+            
+            ArcBase.IsVisible = true;
+            
+            
+            if (IsBilateral) DrawArcBilateral();
+            else DrawArcQuad();
+        }
+        
+        protected override void DrawArc(Path Arc, double value, bool overrideBase, string color = "ThemeAccentBrush"){
```
Comment (mat1jaczyyy) [inline Apollo/Components/GraphDial.cs]: yeah this is p much entirely flipped on its head, ill have to refactor this

File: Apollo/Devices/Copy.cs [hunk 4]
```diff
@@ -203,19 +215,24 @@ class IntTuple {
...
         ConcurrentHashSet<PolyInfo> poly = new ConcurrentHashSet<PolyInfo>();
 
-        public override Device Clone() => new Copy(_time.Clone(), _gate, Pinch, Reverse, Infinite, CopyMode, GridMode, Wrap, (from i in Offsets select i.Clone()).ToList(), Angles.ToList()) {
+        public override Device Clone() => new Copy(_time.Clone(), _gate, Pinch, Reverse, Infinite, CopyMode, GridMode, Wrap, (from i in Offsets select i.Clone()).ToList(), Angles.ToList(), Bilateral) {
             Collapsed = Collapsed,
             Enabled = Enabled
         };
 
-        public Copy(Time time = null, double gate = 1, double pinch = 0, bool reverse = false, bool infinite = false, CopyType copymode = CopyType.Static, GridType gridmode = GridType.Full, bool wrap = false, List<Offset> offsets = null, List<int> angles = null): base("copy") {
+        public Copy(Time time = null, double gate = 1, double pinch = 0, bool reverse = false, bool infinite = false, CopyType copymode = CopyType.Static, GridType gridmode = GridType.Full, bool wrap = false, List<Offset> offsets = null, List<int> angles = null, bool bilateral = false): base("copy") {
```
Comment (mat1jaczyyy) [inline Apollo/Devices/Copy.cs]: pls put bilateral right after pinch in arglist, same in savefile etc

File: Apollo/Components/Dial.xaml (hunk 1 above)
Comment (Brendonovich) [reply Apollo/Components/Dial.xaml]: honestly no idea lol

File: Apollo/Components/GraphDial.cs (hunk 3 above)
Comment (Brendonovich) [reply Apollo/Components/GraphDial.cs]: aight

**Comment by mat1jaczyyy** [comment]
qa pls

**Comment by mat1jaczyyy** [comment]
I'll just merge this so I can work more on it

**Pull request merged**